src/api/app/models/package.rb

```diff
@@ -1010,6 +1010,24 @@ def is_local_link?
     linkinfo && (linkinfo['project'] == project.name)
   end
 
+  def add_containers
+    container_list = []
+    origin_container(local: false).binary_releases.each do |binary_release|
+      container_list += binary_release.media_containers
+    end
+    comment = "add container for #{name}"
+
+    container_list.uniq.each do |container|
+      # create a package container
+      target_package = Package.new(name: name, title: container.title, description: container.description)
+      project.packages << target_package
+      target_package.store(comment: comment)
```

Comment:
This will throw an exception (`.store` uses `.save!`) in case it goes wrong, for instance because of a validation. Are you sure this can't happen?

Reply:
it can only happen when the backed is not reachable IMHO. I am not aware that it is possible to construct something what could fail in an exception...

Reply:
For instance if the package name is using an invalid string?